thunder/executors/torch_autograd.py

```diff
@@ -123,7 +128,18 @@ def detach_if_tensor(t):
             ctx.side_channel["tensors_to_save"] = saved_tensors
             return torch.randn(1, device="meta", requires_grad=True)
         else:
+            if is_differentiable_outputs is None:
+                # Default to original behavior of marking all outputs as differentiable.
+                is_differentiable_outputs = tuple(True for _ in flat_output)
+
             ctx.save_for_backward(*saved_tensors)
+
+            assert len(flat_output) == len(is_differentiable_outputs)
+            filter_non_differentiable = [
```

Comment:
Why do you prefer to propagate "differentiable" output information to invert this right before the `ctx.mark_non_differentiable` call? It seems more straightforward, especially for those who know about `mark_non_differentiable` to propagate "non-differentiable" info.

Reply:
While developing and debugging, `is_differentiable` info seemed to make a lot more sense while comparing against the trace. I am open to propagating `non-differentiable` info instead, wdyt?